create.ts

```diff
@@ -238,41 +239,63 @@ const sharedSwcOption: SwcOptions = {
   module: { type: 'commonjs' }
 };
 
+const sharedEsmLikeFiles = {
+  'implementation.js': '\'use strict\';\nmodule.exports = require(\'./entry.js\').implementation;\n',
+  'implementation.d.ts': 'import type Entry from \'./entry\';\ndeclare const _default: typeof Entry[\'implementation\'];\nexport default _default;\n',
+  'polyfill.js': '\'use strict\';\nmodule.exports = require(\'./entry.js\').polyfill;\n',
+  'polyfill.d.ts': 'import type Entry from \'./entry\';\ndeclare const _default: typeof Entry[\'polyfill\'];\nexport default _default;\n',
+  'shim.js': '\'use strict\';\nmodule.exports = require(\'./entry.js\').shim;\n',
+  'shim.d.ts': 'import type Entry from \'./entry\';\ndeclare const _default: typeof Entry[\'shim\'];\nexport default _default;\n',
+  'auto.js': '\'use strict\';\n/* noop */\n',
+  'auto.d.ts': 'export {}\n',
+  'index.js': '\'use strict\';\nmodule.exports = require(\'./entry.js\').index();\n',
+  'index.d.ts': 'import type Entry from \'./entry\';\ndeclare const _default: typeof Entry[\'implementation\'];\nexport default _default;\n'
+};
+
 async function createEsShimLikePackage(
   packageName: string,
   extraDependencies: Record<string, string> = {},
   minimumNodeVersion = '>=12.4.0'
 ) {
-  const entryContent = await fsPromises.readFile(
-    path.join(__dirname, 'packages', 'data', 'es-shim-like', 'src', `${packageName}.ts`),
-    { encoding: 'utf-8' }
-  );
-  const { code } = await transform(
-    entryContent,
-    sharedSwcOption
-  );
+  const entryPath = path.join(__dirname, 'packages', 'data', 'es-shim-like', 'src', `${packageName}.ts`);
+  const pkgPath = path.join(__dirname, 'packages/generated', packageName);
+
+  const program = ts.createProgram([entryPath], {
+    declaration: true,
+    emitDeclarationOnly: true,
+    outDir: pkgPath,
+    moduleResolution: ts.ModuleResolutionKind.Bundler,
+    target: ts.ScriptTarget.ES2018,
+    lib: ['ES2018']
+  });
+
+  let dtsText: string | undefined;
+
+  program.emit(undefined, (_fileName, text) => { dtsText = text; }, undefined, true);
+
+  const entryContent = await fsPromises.readFile(entryPath, 'utf-8');
+  const { code } = await transform(entryContent, sharedSwcOption);
```

Comment:
I'm thinking should we just use `typescript` instead in case we're generating `types` anyway here then.

Reply:
@JounQin I use swc simply because we also use `@swc-node/register` so we can share the same `@swc/core`.

Reply:
> @JounQin I use swc simply because we also use `@swc-node/register` so we can share the same `@swc/core`. Also the minification.

I'm not sure to understand the statements here, are we accepting to replace `swc` to `tsc` here?